pkg/auth/cookie_manager.go

```diff
@@ -76,22 +75,37 @@ func (c CookieManager) SetTokenCookies(ctx context.Context, writer http.Response
 		return errors.Errorf(ErrTokenNil, "Attempting to set cookies with nil token")
 	}
 
-	jwtCookie, err := NewSecureCookie(accessTokenCookieName, token.AccessToken, c.hashKey, c.blockKey)
+	atCookie, err := NewSecureCookie(accessTokenCookieName, token.AccessToken, c.hashKey, c.blockKey)
 	if err != nil {
-		logger.Errorf(ctx, "Error generating encrypted JWT cookie %s", err)
+		logger.Errorf(ctx, "Error generating encrypted accesstoken cookie %s", err)
 		return err
 	}
-	http.SetCookie(writer, &jwtCookie)
+
+	http.SetCookie(writer, &atCookie)
+
+	if idTokenRaw, converted := token.Extra(idTokenExtra).(string); converted {
+		idCookie, err := NewSecureCookie(idTokenCookieName, idTokenRaw, c.hashKey, c.blockKey)
+		if err != nil {
+			logger.Errorf(ctx, "Error generating encrypted id token cookie %s", err)
+			return err
+		}
+
+		http.SetCookie(writer, &idCookie)
```

Comment:
why is it now required to set an idCookie?

Reply:
Per OIdC spec, it's a required response parameter: https://openid.net/specs/openid-connect-core-1_0.html#TokenResponse

So we are guaranteed to get it from any OIdC provider after a successful authN...